packages/terra-dropdown-button/src/SplitButton.jsx

```diff
@@ -149,7 +149,7 @@ class SplitButton extends React.Component {
   handleCaretKeyDown(event) {
     if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
       // In FireFox active styles don't get applied onKeyDown
-      this.setState({ caretIsActive: true, openedViaKeyboard: true });
+      this.setState(prevState => ({ caretIsActive: true, openedViaKeyboard: true, isOpen: !prevState.isOpen }));
```

Comment:
Setting isOpen in keydown will cause dropdown to close immediately as we leave space OR return key after holding it for some time( long press ) . To adrress this issue `openDropdown()` is being handled only in `onKeyUp` event. 

So with this change we can see that dropdown gets open on long press of space OR return key and it closes the dropdown as soon as we leave the key.
![dropdownbutton](https://user-images.githubusercontent.com/12869809/84246054-a6de1a00-ab23-11ea-8b8b-586080f79ff3.gif)

Reply:
Removed setState and fixed with preventDefault 4170c40